### DISCUSSION TITLE
Редактирование источника в Селектах для таблиц Циклов

### DISCUSSION THREAD

**Discussion by EKhorikov** [totumonline/totum-ru-issues-and-discussions#60] [Как это сделать?] [answered]
Подскажите, а создание новой записи через [+] возможно, если таблица селекта является таблицей Циклов, с заблокированными для редактирования ячейками? При нажатии на плюс (добавить новую запись) окно открывается, только сохранить заполненное окно, естественно, нельзя.

<img width="643" alt="Снимок экрана 2022-01-30 в 14 54 52" src="https://user-images.githubusercontent.com/55755565/151698555-a2f8fc7d-a4cd-4d3c-b2d6-384e2894db6f.png">

edit by APolunin

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Да, вижу, что если все заблокировано то там нет `Сохранить`. Если разблокировано то произойдет переход в цикл при сохранении. Я думаю, что мы **уберем** из списка таблиц для изменения таблицы циклов и временные, а также расчетные если поле находится вне цикла.

Сделать добавление и изменение в таких случаях можно через `panelButton`, прописав туда, что должно происходить

**Reply by EKhorikov** [reply]
Это будет работать так же через [+]? 
Опишите подробней, где panelButton применить?

**Reply by ttm-alexeypolunin** [reply]
Например мы хотим добавить строку в таблицу источник и присвоить полю Селекта значение с этой добавленной строкой.

В форматировании поля селекта (только звездочки уберите, они здесь для корректного отображения встроенного кода):

```
p1=: panelButton(code: $code; text: "ADD NEW"; refresh: true)

*   ```code:totum
*   a1=: insert(table: 'source'; field: 'name' = "from panel button"; inserts: "inserts"; log: true)
*
*   a2=: set(table: $#ntn; field: 'select_field' = $#inserts[0]; where: 'id' = #id; log: true)
*   ```

```
